### ISSUE TITLE
Support for `extern enum`s?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#10323]
[Filed as part of issue #9222]

Should Chapel support extern declarations of enums as a means of referring to C enums within Chapel, as in:

```chapel
extern enum { red, green, blue};
```

Pros
----
* seems attractive

Cons
-----
* will likely require effort, particularly as our enum semantics and implementation diverge from C's (e.g., #10026, #10307).

**Comment by ben-albrecht** [comment]
One use-case for this feature:

When using the `BLAS` module with `openblas` as the backend implementation, the compilation generates several warnings about enum conversions, since `BLAS` uses internal enums to mirror the extern enums. Supporting extern enums would enable us to silence those warnings.

**Comment by lydia-duncan** [comment]
Maybe we would want to do this at the same time as enabling the export of Chapel enums?  The translation between the two would hopefully work both directions.

**Comment by bradcray** [comment]
I don't think this would just fall out — exporting and externing end up requiring pretty different support.  `extern`s permit us to squash codegen-ing of things and use C definitions instead.  `export`s require us to figure out how to represent Chapel types in C code (and I believe that'll have some interesting challenges given planned directions for enums like #10307.  I.e., I'd open a separate issue for the `export` case.
